File: MyMedTimer/Models/Medication.swift
import Foundation
import SwiftData

@Model
final class Medication {
    var id: UUID = UUID()
    var name: String = ""
    var dosage: String = ""
    var colorHex: String = "#FF6B6B"
    var alertStyle: String = "gentle"
    var isPRN: Bool = false
    var minIntervalMinutes: Int = 0
    var isActive: Bool = true
    var createdAt: Date = Date()

    @Relationship(deleteRule: .cascade, inverse: \ScheduleTime.medication)
    var scheduleTimes: [ScheduleTime] = []

    @Relationship(deleteRule: .cascade, inverse: \DoseLog.medication)
    var doseLogs: [DoseLog] = []

    init(name: String, dosage: String, colorHex: String = "#FF6B6B", alertStyle: String = "gentle") {
        self.name = name
        self.dosage = dosage
        self.colorHex = colorHex
        self.alertStyle = alertStyle
    }
}

File: android/core/domain/src/main/kotlin/com/nateb/mymedtimer/domain/model/Medication.kt
package com.nateb.mymedtimer.domain.model

import java.time.Instant

data class Medication(
    val id: String,
    val name: String,
    val dosage: String,
    val colorHex: String = "#FF6B6B",
    val alertStyle: String = "gentle",
    val isPRN: Boolean = false,
    val minIntervalMinutes: Int = 0,
    val isActive: Boolean = true,
    val createdAt: Instant = Instant.now(),
    val scheduleTimes: List<ScheduleTime> = emptyList(),
    val doseLogs: List<DoseLog> = emptyList()
)
